src/server/metrics.rs

```diff
@@ -81,19 +132,69 @@ lazy_static! {
+        "tikv_gcworker_gc_tasks_vec",
+        "Counter of gc tasks processed by gc_worker",
+        &["task"]
+    )
+    .unwrap();
+    pub static ref GC_GCTASK_FAIL_COUNTER_VEC: IntCounterVec = register_int_counter_vec!(
+        "tikv_gcworker_gc_task_fail_vec",
+        "Counter of gc tasks that is failed",
+        &["task"]
+    )
+    .unwrap();
+    pub static ref RESOLVE_STORE_COUNTER: IntCounterVec = register_int_counter_vec!(
+        "tikv_server_resolve_store_total",
+        "Total number of resolving store",
+        &["type"]
+    )
+    .unwrap();
+    pub static ref GRPC_MSG_FAIL_COUNTER_VEC: IntCounterVec = register_int_counter_vec!(
+        "tikv_grpc_msg_fail_total",
+        "Total number of handle grpc message failure",
+        &["type"]
+    )
+    .unwrap();
+    pub static ref GC_KEYS_COUNTER_VEC: IntCounterVec = register_int_counter_vec!(
+        "tikv_gcworker_gc_keys",
+        "Counter of keys affected during gc",
+        &["cf", "tag"]
+    )
+    .unwrap();
+    pub static ref GRPC_MSG_HISTOGRAM_VEC: HistogramVec = register_histogram_vec!(
         "tikv_grpc_msg_duration_seconds",
         "Bucketed histogram of grpc server messages",
         &["type"],
         exponential_buckets(0.0005, 2.0, 20).unwrap()
     )
     .unwrap();
-    pub static ref GRPC_MSG_FAIL_COUNTER: GrpcMsgFailCounterVec = register_static_int_counter_vec!(
-        GrpcMsgFailCounterVec,
-        "tikv_grpc_msg_fail_total",
-        "Total number of handle grpc message failure",
-        &["type"]
+}
+
+lazy_static! {
+    pub static ref GRPC_MSG_HISTOGRAM_STATIC: GrpcMsgHistogramVec =
+        auto_flush_from!(GRPC_MSG_HISTOGRAM_VEC, GrpcMsgHistogramVec);
+    pub static ref GC_COMMAND_COUNTER_VEC_STATIC: GcCommandCounterVec =
+        auto_flush_from!(GC_COMMAND_COUNTER_VEC, GcCommandCounterVec);
+    pub static ref SNAP_TASK_COUNTER_STATIC: SnapTaskCounterVec =
+        auto_flush_from!(SNAP_TASK_COUNTER, SnapTaskCounterVec);
+    pub static ref GC_GCTASK_COUNTER_STATIC: GcTaskCounterVec =
+        auto_flush_from!(GC_GCTASK_COUNTER_VEC, GcTaskCounterVec);
+    pub static ref GC_GCTASK_FAIL_COUNTER_STATIC: GcTaskFailCounterVec =
+        auto_flush_from!(GC_GCTASK_FAIL_COUNTER_VEC, GcTaskFailCounterVec);
+    pub static ref RESOLVE_STORE_COUNTER_STATIC: ResolveStoreCounterVec =
+        auto_flush_from!(RESOLVE_STORE_COUNTER, ResolveStoreCounterVec);
+    pub static ref GRPC_MSG_FAIL_COUNTER: GrpcMsgFailCounterVec =
+        auto_flush_from!(GRPC_MSG_FAIL_COUNTER_VEC, GrpcMsgFailCounterVec);
+    pub static ref GC_KEYS_COUNTER_STATIC: GcKeysCounterVec =
+        auto_flush_from!(GC_KEYS_COUNTER_VEC, GcKeysCounterVec);
+}
+
+lazy_static! {
+    pub static ref SEND_SNAP_HISTOGRAM: Histogram = register_histogram!(
+        "tikv_server_send_snapshot_duration_seconds",
+        "Bucketed histogram of server send snapshots duration",
+        exponential_buckets(0.05, 2.0, 20).unwrap()
```

Comment:
why do we have two sections of metrics? (another at L122)

Reply:
> why do we have two sections of metrics? (another at L122)


The original version have two sections, I guest they don't want one section to be too long